### DISCUSSION TITLE
AI 聊天服务 AzureOpenAI 範例問題

### DISCUSSION THREAD

**Discussion by james-chao** [dotnetcore/BootstrapBlazor#2455] [Q&A] [answered]
請問大佬，BB 網站的新組件並沒有AI 聊天服务 AzureOpenAI的相關資訊範例，只有通过调用 IAzureOpenAIService 服务方法 GetChatCompletionsAsync 进行聊天对话 這段話，請問要如何運用AI 聊天服务 AzureOpenAI 這個組件，感謝!

**Comment by densen2014** [comment]
key部署一下直接用，参考源码

**Comment by ArgoZhang** [comment, marked as answer]
@james-chao 这个组件比较特殊，是需要开通 `Azure OpenAI` 服务的，需要配置如下：

appsettings.json
```
"AzureOpenAIOption": {
  "Endpoint": "<Endpoint>",
  "Key": "<Key>",
  "DeploymentName": "<DeploymentName>"
}
```

需要在配置文件中配置这三个配置项，其值为 `azure` 服务提供，然后就是在代码中获得注入服务进行交互即可

https://github.com/dotnetcore/BootstrapBlazor/blob/main/src/BootstrapBlazor.Server/Components/Pages/Chats.razor.cs#L17C16-L17C16
